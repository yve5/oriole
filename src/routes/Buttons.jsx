import React, { Fragment } from 'react';
import Confetti from 'canvas-confetti';

const buttons = [
  { title: 'Primary', suffix: 'primary' },
  { title: 'Secondary', suffix: 'outline-secondary' },
  { title: 'Success', suffix: 'success' },
  { title: 'Danger', suffix: 'danger' },
  { title: 'Warning', suffix: 'warning' },
  { title: 'Info', suffix: 'info' },
  { title: 'Light', suffix: 'outline-light' },
  { title: 'Dark', suffix: 'dark' },
  { title: 'Link', suffix: 'link' },
  // { title: 'Outline Primary', suffix: 'outline-primary' },
  // { title: 'Outline Secondary', suffix: 'outline-secondary' },
  // { title: 'Outline Success', suffix: 'outline-success' },
  // { title: 'Outline Danger', suffix: 'outline-danger' },
  // { title: 'Outline Warning', suffix: 'outline-warning' },
  // { title: 'Outline Info', suffix: 'outline-info' },
  // { title: 'Outline Light', suffix: 'outline-light' },
  // { title: 'Outline Dark', suffix: 'outline-dark' },
  // { title: 'Outline Link', suffix: 'outline-link' },
];

const Buttons = () => (
  <div className="card">
    <ul className="list-group list-group-flush">
      <li className="list-group-item">
        {buttons.map(({ title, suffix }) => (
          <button
            className={`btn btn-${suffix} me-2 mb-2`}
            data-testid="button-demo"
            onClick={() =>
              Confetti({
                particleCount: 100,
                origin: { y: 0.6 },
                spread: 70,
              })
            }
            type="button"
            key={suffix}
          >
            {title}
          </button>
        ))}
      </li>
      <li className="list-group-item bg-light-subtle">
        <code>
          {buttons.map(({ title, suffix }) => (
            <Fragment key={suffix}>
              &lt;button className=&quot;btn btn-{suffix}&quot;
              type=&quot;button&quot;&gt;
              {title}&lt;/button&gt;
              <br />
            </Fragment>
          ))}
        </code>
      </li>
    </ul>
  </div>
);

export default Buttons;
