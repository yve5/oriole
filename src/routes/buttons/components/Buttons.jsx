import React, { Fragment } from 'react';
import Confetti from 'canvas-confetti';
import { clsx } from 'clsx';

import { ORIOLE_BUTTONS } from '../resources/constants';

const ButtonLayer = ({ className = '' }) => (
  <div className={clsx('card', 'mb-4')}>
    <ul className={clsx('list-group', 'list-group-flush')}>
      <li className={clsx('list-group-item')}>
        {ORIOLE_BUTTONS.map(({ title, suffix }) => (
          <button
            key={suffix}
            type="button"
            className={`btn btn-${suffix} me-2 mb-2 ${className}`}
            onClick={() =>
              Confetti({ spread: 70, origin: { y: 0.6 }, particleCount: 100 })
            }
          >
            {title}
          </button>
        ))}
      </li>
      <li className={clsx('list-group-item', 'bg-light-subtle')}>
        <code>
          {ORIOLE_BUTTONS.map(({ title, suffix }) => (
            <Fragment key={suffix}>
              &lt;button className=&quot;btn btn-{suffix} {className}&quot;
              type=&quot;button&quot;&gt;
              {title}&lt;/button&gt;
              <br />
            </Fragment>
          ))}
        </code>
      </li>
    </ul>
  </div>
);

const Buttons = () => (
  <>
    <ButtonLayer className="btn-sm" />
    <ButtonLayer className="btn-sm disabled" />
  </>
);

export default Buttons;
