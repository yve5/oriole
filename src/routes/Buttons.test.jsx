import React from 'react';
import { expect, test, vi } from 'vitest';
import { act, create } from 'react-test-renderer';

import Buttons from './Buttons';

test('should match expected snapshot', () => {
  HTMLCanvasElement.prototype.getContext = vi.fn(() => ({
    arc: vi.fn(),
    beginPath: vi.fn(),
    clearRect: vi.fn(),
    closePath: vi.fn(),
    fill: vi.fn(),
    lineTo: vi.fn(),
    moveTo: vi.fn(),
    rotate: vi.fn(),
    restore: vi.fn(),
    save: vi.fn(),
    scale: vi.fn(),
    translate: vi.fn(),
  }));

  const component = create(<Buttons />);

  act(() => {
    component.update(<Buttons />);
  });

  act(() => {
    component.root
      .findAllByProps({ 'data-testid': 'button-demo' })[0]
      .props.onClick();
  });

  expect(component.toJSON()).toMatchSnapshot();
});
