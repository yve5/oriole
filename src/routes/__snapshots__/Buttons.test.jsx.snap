// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match expected snapshot 1`] = `
<div
  className="card"
>
  <ul
    className="list-group list-group-flush"
  >
    <li
      className="list-group-item"
    >
      <button
        className="btn btn-primary me-2 mb-2"
        data-testid="button-demo"
        onClick={[Function]}
        type="button"
      >
        Primary
      </button>
      <button
        className="btn btn-outline-secondary me-2 mb-2"
        data-testid="button-demo"
        onClick={[Function]}
        type="button"
      >
        Secondary
      </button>
      <button
        className="btn btn-success me-2 mb-2"
        data-testid="button-demo"
        onClick={[Function]}
        type="button"
      >
        Success
      </button>
      <button
        className="btn btn-danger me-2 mb-2"
        data-testid="button-demo"
        onClick={[Function]}
        type="button"
      >
        Danger
      </button>
      <button
        className="btn btn-warning me-2 mb-2"
        data-testid="button-demo"
        onClick={[Function]}
        type="button"
      >
        Warning
      </button>
      <button
        className="btn btn-info me-2 mb-2"
        data-testid="button-demo"
        onClick={[Function]}
        type="button"
      >
        Info
      </button>
      <button
        className="btn btn-outline-light me-2 mb-2"
        data-testid="button-demo"
        onClick={[Function]}
        type="button"
      >
        Light
      </button>
      <button
        className="btn btn-dark me-2 mb-2"
        data-testid="button-demo"
        onClick={[Function]}
        type="button"
      >
        Dark
      </button>
      <button
        className="btn btn-link me-2 mb-2"
        data-testid="button-demo"
        onClick={[Function]}
        type="button"
      >
        Link
      </button>
    </li>
    <li
      className="list-group-item bg-light-subtle"
    >
      <code>
        &lt;button className="btn btn-
        primary
        " type="button"&gt;
        Primary
        &lt;/button&gt;
        <br />
        &lt;button className="btn btn-
        outline-secondary
        " type="button"&gt;
        Secondary
        &lt;/button&gt;
        <br />
        &lt;button className="btn btn-
        success
        " type="button"&gt;
        Success
        &lt;/button&gt;
        <br />
        &lt;button className="btn btn-
        danger
        " type="button"&gt;
        Danger
        &lt;/button&gt;
        <br />
        &lt;button className="btn btn-
        warning
        " type="button"&gt;
        Warning
        &lt;/button&gt;
        <br />
        &lt;button className="btn btn-
        info
        " type="button"&gt;
        Info
        &lt;/button&gt;
        <br />
        &lt;button className="btn btn-
        outline-light
        " type="button"&gt;
        Light
        &lt;/button&gt;
        <br />
        &lt;button className="btn btn-
        dark
        " type="button"&gt;
        Dark
        &lt;/button&gt;
        <br />
        &lt;button className="btn btn-
        link
        " type="button"&gt;
        Link
        &lt;/button&gt;
        <br />
      </code>
    </li>
  </ul>
</div>
`;
